worker/raft/worker.go

```diff
@@ -292,7 +346,22 @@ func (w *Worker) loop(raftConfig *raft.Config) (loopErr error) {
 			// the local node was removed from the cluster
 			// configuration, causing it to shutdown.
 			return errors.New("raft shutdown")
+		case <-noLeaderCheck:
+			noLeaderCheck = w.config.Clock.After(noLeaderFrequency)
+			if r.State() == raft.Leader {
+				lastContact = w.config.Clock.Now()
+				continue
+			}
+			var zeroTime time.Time
+			if r.LastContact() != zeroTime {
+				lastContact = r.LastContact()
+			}
+			if w.config.Clock.Now().After(lastContact.Add(w.config.NoLeaderTimeout)) {
```

Comment:
Should we do

now := w.config.Clock.Now()

at the top of this block of code and use that

Reply:
I'll do this.

Reply:
Actually, the Clock.After channel returns the time, so I used that.